crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -1687,23 +1682,22 @@ impl ScheduleGraph {
     /// If the graph contain cycles, then an error is returned.
     fn topsort_graph(
         &self,
-        graph: &DiGraphMap<NodeId, ()>,
+        graph: &DiGraph,
         report: ReportCycles,
     ) -> Result<Vec<NodeId>, ScheduleBuildError> {
         // Tarjan's SCC algorithm returns elements in *reverse* topological order.
-        let mut tarjan_scc = TarjanScc::new();
         let mut top_sorted_nodes = Vec::with_capacity(graph.node_count());
         let mut sccs_with_cycles = Vec::new();
 
-        tarjan_scc.run(graph, |scc| {
+        for scc in graph.iter_scc() {
             // A strongly-connected component is a group of nodes who can all reach each other
             // through one or more paths. If an SCC contains more than one node, there must be
             // at least one cycle within them.
+            top_sorted_nodes.extend_from_slice(&scc);
             if scc.len() > 1 {
-                sccs_with_cycles.push(scc.to_vec());
+                sccs_with_cycles.push(scc);
             }
-            top_sorted_nodes.extend_from_slice(scc);
-        });
+        }
```

Comment:
Same here, I think this could be done with something like
```rs
top_sorted_nodes.extend(graph.iter_scc().map(|scc| { if scc.len() > 1 { .. }; scc }).flatten());
```
But unsure, because it needs an `IntoIterator`, maybe use `TarjanScc` directly